crates/api_server/src/server.rs

```diff
@@ -105,34 +87,18 @@ impl NodeServerBuilder {
             .set_rpc_middleware(RpcServiceBuilder::new().rpc_logger(100));
 
         let server = server_builder.build(addr).await.unwrap();
-        let rpc = self.rpc.clone();
-        self.server_futs
-            .push(Box::pin(async move { server.start(rpc) }));
-    }
-
-    pub async fn run(self) -> NodeServerHandle {
-        let handles = futures::future::join_all(self.server_futs).await;
-        NodeServerHandle { handles }
+        let rpc = Self::default_rpc(self.node);
+        NodeServer(Box::new(move || server.start(rpc)))
```

Comment:
It's indeed quite a strange design decision by jsonrpsee to handle spawning a task by themselves (and not exposing an API to do so manually), so probably it makes sense to write a comment explaining why we are doing this weird thing, e.g. something like:

// `jsonrpsee` does `tokio::spawn` within `start` method, so we cannot invoke it here, as this method
// should only _build_ the server. This way we delay the launch until the `NodeServer::run` is invoked.